src/Search.js

```diff
@@ -10,18 +10,11 @@ class Search extends Component {
     this.handleKeyPress = this.handleKeyPress.bind(this)
   }
 
-  state = {
-    text: ''
-  }
-
   handleClick() {
     this.props.onShowSearch()
   }
 
   handleChange(e) {
-    this.setState({
-      text: e.target.value
-    })
```

Comment:
You're currently holding the same state in 2 places - inside this component + inside App.state.searchText - you should only have one source of truth for the state, which means removing it here when you push it up to the parent.